Sources/MapboxCoreNavigation/SimulatedLocationManager.swift

```diff
@@ -160,9 +162,9 @@ open class SimulatedLocationManager: NavigationLocationManager {
                                   course: newCoordinate.direction(to: lookAheadCoordinate).wrap(min: 0, max: 360),
                                   speed: currentSpeed,
                                   timestamp: Date())
-        
+
         self.simulatedLocation = location
-        
+
         delegate?.locationManager?(self, didUpdateLocations: [location])
```

Comment:
Since `tick` is now executed on another thread, should this delegate call be explicitly done on `main`?

Default `CLLocationManager` captures the thread it was instantiated on for location update calls, but our implementation forced a main thread

Reply:
It may make sense to get back a delegate call to the main thread to minimize potential side effects in execution.
However, I doubt that anything downstream should rely on this call being executed on the main thread.

Reply:
Now I remember why I left the delegate call on the same queue as `tick` - out tests heavily rely on synchronous delegate call inside `tick`.

Reply:
IMO it's a risky change. What if we add a nullable parameter to the manager with the queue on which the callback should be called and keep it `nil` in tests to make calls synchronous while having the default value of `main` queue?

Reply:
But I believe the point of this PR is to make it non-main thread by default... As Alex said, our tests are failing if we introduce explicit main thread call, so there's a good chance the same will happen with user code. We could instead document that delegate will be called on a background instead, so user will be able to manage threading himself if he needs it.